### PR TITLE
feat(atomic): added live region for screen readers 

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-244

Tested with:
* JAWS
* NVDA
* VoiceOver (iOS)
* TalkBack
* VoiceOver (Mac OS)

The region notifies screen reader users of the following events:
* `Loading new results`
* `Results {{first}}-{{last}} of {{total}}` (re-used that string due to the amount of translations for it)
* `No results`

There's a few caveats with this:
1. Screen readers will always read out "Loading new results" before reading "Results #-# of #", even if results finished loading long before it started saying "Loading new results".
2. Hearing "Results #-# of #" or "No results" is a bit confusing on its own when you're interacting with facets or the sort button. Do you think I should create new strings for those that use verbs in the continuous tense? @dlutzCoveo @npushkarskii @johnnyCoveo

### CODE REVIEW

File: packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+    this.querySummary = buildQuerySummary(this.bindings.engine);
+  }
+
+  public render() {
+    return (
+      <Host
+        role="status"
+        aria-live="polite" // redundant, but recommended in Mozilla's doc
+        style={{position: 'absolute', right: '10000px'}}
```
Comment (btaillon) [inline packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx:65]: `transform: scale(0)` is detected by VoiceOver on iOS as meaning the element doesn't exist, preventing the region from being read.

File: packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+    this.querySummary = buildQuerySummary(this.bindings.engine);
+  }
+
+  public render() {
+    return (
+      <Host
+        role="status"
+        aria-live="polite" // redundant, but recommended in Mozilla's doc
```
Comment (btaillon) [inline packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx:64]: [Source](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions#:~:text=the%20current%20status.-,to%20maximize%20compatibility%2C%20add%20a%20redundant%20aria-live%3D%22polite%22%20when%20using%20this%20role.,-alert)

File: packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  private get searchSummary() {
+    if (!this.searchStatusState.firstSearchExecuted) {
+      return '';
+    }
+    if (this.searchStatusState.isLoading) {
+      return this.bindings.i18n.t('loading-results');
+    }
+    if (!this.searchStatusState.hasResults) {
+      return this.bindings.i18n.t('no-results');
```
Comment (samisayegh) [inline packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx:45]: When no results are caused by an error due to the connection, the token etc. what would happen?
Should Atomic's aria-live announce a more descriptive message in these situations?

File: packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx (hunk 2 above)
Comment (samisayegh) [reply packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx:64]: I would keep the attribute but I think we can remove the comment,

**Review by samisayegh — APPROVED** [review]

File: packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx (hunk 3 above)
Comment (olamothe) [reply packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx:45]: Hum good point ... 🤔 Do we handle that in JSUI ?

**Review by olamothe — APPROVED** [review]

File: packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx (hunk 3 above)
Comment (samisayegh) [reply packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx:45]: Yep, there is a separate [message](https://github.com/coveo/search-ui/blob/master/src/ui/AriaLive/AriaLive.ts#L39) for errors.

Comment (btaillon) [reply packages/atomic/src/components/atomic-search-interface/atomic-aria-live.tsx:45]: Good point, I'll handle that.

**Review by npushkarskii — APPROVED** [review]

**Pull request closed without merging**